connection.go

```diff
@@ -554,9 +554,23 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
 // go sql library but is exported to clients who can make use of this
 // capability explicitly.
 func (sc *snowflakeConn) WaitForQueryCompletion(ctx context.Context, qid string) error {
+	// if the query is already done, we dont need to wait for it 
+	complete, err := sc.checkIfComplete(ctx, qid) 
+	if complete {
+		return err
+	}
```

Comment:
I kind of get what's going on here from reading the code in `checkIfComplete`, but this feels a bit weird to me.

Reply:
I will reword it a bit so its easier to follow